microovn/ovn/services.go

```diff
@@ -0,0 +1,84 @@
+package ovn
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/canonical/microcluster/state"
+	"github.com/canonical/microovn/microovn/database"
+	"github.com/canonical/microovn/microovn/node"
+)
+type SrvName string
+
+const (
+	SrvChassis SrvName = "chassis"
+	SrvCentral SrvName = "central"
+	SrvSwitch  SrvName = "switch"
+)
+var ServiceNames = []SrvName{SrvChassis,SrvCentral,SrvSwitch}
+
+func DisableService(s *state.State, service string) error {
+	err := s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
```

Comment:
We need to do validation of the `service` string prior to this call.

Reply:
why, the service name gets validated within this call by checking it against the database?

Reply:
It is a precaution in the event of vulnerable code in our library dependencies.

Our backing database is SQL-based, and history is riddled with security incidents rooted in SQL injection attacks.

While our code has solid mitigations for that by making use of [prepared statements](https://go.dev/doc/database/prepared-statements) with place holders for variables:
https://github.com/canonical/microovn/blob/a99570ddbcffe6acbd76698a424c78d3cca49773/microovn/database/service.mapper.go#L51-L55

which should avoid such issues, I'd rather be safe than sorry by validating the data before exposing it to our data store.

Reply:
I did look and check that the input is sanitized for the database queries, I shall validate the input in the API, though I stand by my opinion on it being somewhat unneccicary